packages/shared/src/hooks/streaks/useReadingStreak.ts

```diff
@@ -1,27 +1,49 @@
-import { useQuery } from '@tanstack/react-query';
+import { useQuery, useQueryClient } from '@tanstack/react-query';
+import { useRef } from 'react';
 import { generateQueryKey, RequestKey, StaleTime } from '../../lib/query';
 import { getReadingStreak, UserStreak } from '../../graphql/users';
 import { useAuthContext } from '../../contexts/AuthContext';
-import { useStreakExperiment } from './useStreakExperiment';
+import { useActions } from '../useActions';
+import { ActionType } from '../../graphql/actions';
+import useDebounce from '../useDebounce';
 
 interface UserReadingStreak {
   streak: UserStreak;
   isLoading: boolean;
-  isEnabled: boolean;
+  shouldShowPopup: boolean;
+  checkReadingStreak: () => Promise<void>;
 }
 
 export const useReadingStreak = (): UserReadingStreak => {
   const { user } = useAuthContext();
-  const { shouldShowStreak } = useStreakExperiment();
   const { data: streak, isLoading } = useQuery(
     generateQueryKey(RequestKey.UserStreak, user),
     getReadingStreak,
-    { enabled: shouldShowStreak, staleTime: StaleTime.Default },
+    { enabled: true, staleTime: StaleTime.Default },
   );
+  const { checkHasCompleted } = useActions();
+  const hasReadToday =
+    new Date(streak?.lastViewAt).getDate() === new Date().getDate();
+  const userStreakQueryKeyRef = useRef<unknown[]>();
+  const queryClient = useQueryClient();
+
+  const [clearQueries] = useDebounce(async () => {
```

Comment:
Why do we debounce here?

Reply:
Kept this as it was in useIncrementReadingRank since it was invalidating user streaks query as well, and totally removed useIncrementReadingRank https://github.com/dailydotdev/apps/pull/3091/files#diff-2ca4b5627b297d8adac0ec5fa73626249c395ceb53225a067d128ed3fad2a6d5